source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -180,38 +190,115 @@ def _paint(self):
 				# When the focus overlaps the navigator object, which is usually the case,
 				# show a different highlight style.
 				# Focus is in contextRects, do not show the standalone focus highlight.
-				contextRects.pop(Context.FOCUS)
+				_ = contextRects.pop(Context.FOCUS)
 				# Navigator object might be in contextRects as well
-				contextRects.pop(Context.NAVIGATOR, None)
+				_ = contextRects.pop(Context.NAVIGATOR, None)
 				context = Context.FOCUS_NAVIGATOR
 			contextRects[context] = rect
+		return contextRects
+
+	def _mapRectToClient(self, rect: RectLTRB, style: HighlightStyle) -> RectLTRB | None:
+		"""
+		Transforms a screen rectangle to a client rectangle ready for drawing.
+		Applies: Intersection with Window -> Logical mapping -> Client mapping -> Style Margin.
+		Returns None if mapping fails.
+		"""
+		# Before calculating logical coordinates,
+		# make sure the rectangle falls within the highlighter window
+		rect = rect.intersection(self.location)
+
+		try:
+			rect = rect.toLogical(self.handle)
+			rect = rect.toClient(self.handle)
+			rect = rect.expandOrShrink(style.margin)
+			return rect
```

Comment:
I think my personal preference would be for this to be written more like this
```suggestion
			return rect.toLogical(self.handle) \
				.toClient(self.handle) \
				.expandOrShrink(style.margin)
```